lib/workers/repository/update/pr/changelog/bitbucket/index.ts

```diff
@@ -0,0 +1,76 @@
+import is from '@sindresorhus/is';
+import changelogFilenameRegex from 'changelog-filename-regex';
+import { logger } from '../../../../../../logger';
+import { PagedSourceResultsSchema } from '../../../../../../modules/platform/bitbucket/schema';
+import { BitbucketHttp } from '../../../../../../util/http/bitbucket';
+import { joinUrlParts } from '../../../../../../util/url';
+import type {
+  ChangeLogFile,
+  ChangeLogNotes,
+  ChangeLogProject,
+  ChangeLogRelease,
+} from '../types';
+
+export const id = 'bitbucket-changelog';
+const bitbucketHttp = new BitbucketHttp(id);
+
+export async function getReleaseNotesMd(
+  repository: string,
+  apiBaseUrl: string,
+  sourceDirectory?: string
+): Promise<ChangeLogFile | null> {
+  logger.trace('bitbucket.getReleaseNotesMd()');
+
+  const repositorySourceURl = joinUrlParts(
+    apiBaseUrl,
+    `2.0/repositories`,
+    repository,
+    'src'
+  );
+
+  const rootFiles = (
+    await bitbucketHttp.getJson(
+      repositorySourceURl,
+      {
+        paginate: true,
+      },
+      PagedSourceResultsSchema
+    )
+  ).body.values;
+
+  const allFiles = rootFiles.filter((f) => f.type === 'commit_file');
+
+  const files = allFiles.filter((f) => changelogFilenameRegex.test(f.path));
+
+  const changelogFile = files.shift();
+  if (is.nullOrUndefined(changelogFile)) {
+    logger.trace('no changelog file found');
+    return null;
+  }
+
+  if (files.length !== 0) {
+    logger.debug(
+      `Multiple candidates for changelog file, using ${changelogFile.path}`
+    );
+  }
+
+  const fileRes = await bitbucketHttp.get(
+    joinUrlParts(
+      repositorySourceURl,
+      changelogFile.commit.hash,
+      changelogFile.path
+    )
+  );
+
+  const changelogMd = `${fileRes.body}\n#\n##`;
+  return { changelogFile: changelogFile.path, changelogMd };
+}
+
+export function getReleaseList(
+  project: ChangeLogProject,
+  _release: ChangeLogRelease
+): ChangeLogNotes[] {
+  logger.trace('github.getReleaseList()');
+  logger.warn('TODO: implement getReleaseList() for Bitbucket');
```

Comment:
so it only works for changelog files?

Reply:
Correct - it'll work for any of the files supported by [changelogFilenameRegex](https://github.com/renovatebot/renovate/blob/d4a8f6164089379cc94c8502ea89df6206d51940/lib/workers/repository/update/pr/changelog/bitbucket/index.ts#L43).  

To my knowledge, Bitbucket Cloud doesn't have the concept of `releases` like GitHub etc does

Reply:
if there isn't that concept, we should warn. just log a debug or info message.